### PR TITLE
[kola/]README.md: Quick start addition for test writing

### PR DESCRIPTION
**Pull request by ashcrow**
Split off from https://github.com/coreos/coreos-assembler/pull/518

### CODE REVIEW

**Comment by ashcrow** [comment]
:unicorn: :fireworks: Happy PR 1000! :fireworks: :rainbow:

File: kola/README.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
+# Adding Tests
+
+
+## Quick Start
+
+1. Fork and clone this [`mantle` repository](https://github.com/coreos/mantle/)
```
Comment (arithx) [inline kola/README.md]: s,this,the

File: kola/README.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+2. Move into `kola/tests/` and look for package which your test would fit best
```
Comment (arithx) [inline kola/README.md]: maybe `look for the package` and `your test would best fit`?

File: kola/README.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
+# Adding Tests
+
+
```
Comment (arithx) [inline kola/README.md]: Extra line break

File: kola/README.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+=== RUN   podman.noop
+--- PASS: podman.noop (21.08s)
+PASS, output in _kola_temp/qemu-2019-05-08-1535-16606
+# git add/commit/push...
+# Open PR to get the test added!
+```
+
+## Grouping Tests
+
+Sometimes it makes sense to group tests together under a specific package, especially when these tests are related and require the same test parameters. For `kola` it only takes a forwarding function to do testing groups. This forwarding function should take `cluster.TestCluster` as it's only input, and execute running other tests with `cluster.TestCluster.Run()`.
```
Comment (arithx) [inline kola/README.md]: Worth noting that these steps are ran sequentially and on the same machine. As such, this is not recommended for any groups of tests that modify system state.

Additionally, the `FailFast` flag can be enabled during the test registration to skip any remaining steps after a failure has occurred.

File: kola/README.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+        _ "github.com/coreos/mantle/kola/tests/foo"
+<snip/>
+```
+4. Lastly, use $EDITOR on `kola/tests/foo/foo.go` adding in new test groups and tests.
+
+
+
+
+
+# Full Example
```
Comment (arithx) [inline kola/README.md]: Should this be a `##` heading with the files being `###` so they nestle under the `Adding Tests` top level header?

File: kola/README.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+        _ "github.com/coreos/mantle/kola/tests/coretest"
+        _ "github.com/coreos/mantle/kola/tests/crio"
+        _ "github.com/coreos/mantle/kola/tests/docker"
+        _ "github.com/coreos/mantle/kola/tests/etcd"
+        _ "github.com/coreos/mantle/kola/tests/foo"
+<snip/>
+```
+4. Lastly, use $EDITOR on `kola/tests/foo/foo.go` adding in new test groups and tests.
+
+
```
Comment (arithx) [inline kola/README.md]: Lots of extra line breaks

File: kola/README.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+If we look at `podmanBaseTest` it becomes very obvious that it's not a test of it's own, but a group of tests.
+
+```go
+func podmanBaseTest(c cluster.TestCluster) {
+        c.Run("info", podmanInfo)
+        c.Run("resources", podmanResources)
+        c.Run("network", podmanNetworksReliably)
+}
+```
+
```
Comment (arithx) [inline kola/README.md]: Extra line break

File: kola/README.md (hunk 4 above)
Comment (ashcrow) [reply kola/README.md]: Add in this almost verbatim.

**Comment by ashcrow** [comment]
Updated per review.

**Comment by ashcrow** [comment]
Whenever there is enough :+1: this can be merged.

**Pull request merged**